### PR TITLE
chore: 👷 Add Husky

### PR DESCRIPTION
**Pull request by melMass**
Each commits get autoformatted with `prettier`,
could be used for other things.

https://typicode.github.io/husky

### CODE REVIEW

**Comment by melMass** [comment]
I sneaked 6d2b3daf644738d4ef9c5f2dd38c46840eb0220b in, it's not really related but it was to avoid another PR

File: .husky/pre-commit [hunk 1]
```diff
@@ -0,0 +1,13 @@
+#!/bin/sh
+. "$(dirname "$0")/_/husky.sh"
+
+echo "Formating project and committing..."
+FILES=$(git diff --cached --name-only --diff-filter=ACMR | sed 's| |\\ |g')
+[ -z "$FILES" ] && exit 0
+
+# Prettify all selected files
+echo "$FILES" | xargs ./node_modules/.bin/prettier --ignore-unknown --ignore-path .gitignore --write --plugin-search-dir=.
```
Comment (xat) [inline .husky/pre-commit]: @melMass do you think we should limit this to .js, .jsx and maybe .scss files?

Comment (melMass) [reply .husky/pre-commit]: It should be handled by prettier but we can restrict it before if needed

**Comment by xat** [comment]
@melMass i pulled in pretty-quick, this made it a bit easier to use the same file pattern in the hook as in the npm script.

i also changed the npm script from `npm run prettier` to `npm run format`, think that's a more convenient name and added `.scss` to the list of supported files.

after adding `.scss`, i formatted all scss files.

**Comment by melMass** [comment]
Awesome! Very nice! 
I think we can merge?

**Review by xat — APPROVED** [review]
great idea :)

**Pull request merged**